### PR TITLE
nrf52_bsim: Update to support incoming changes to clock driver

### PR DESCRIPTION
**Pull request by aescolar**
Incoming updates to the nRF clock driver require some extra enum
definitions from the nRFx HAL which were not available in HW
model versions before 1.4 => update it.
Also as the nrf52_bsim does not yet support DTS, include a define
for the clock IRQ interrupt number (as the new clock driver will
be using the DTS macro instead of the nRF MDK value.

Signed-off-by: Alberto Escolar Piedras <alpi@oticon.com>

Blocks #12531

### CODE REVIEW

**Review by anangl — APPROVED** [review]
I don't entirely understand the need for this complex condition check.
Wouldn't just `#if !defined(DT_NORDIC_NRF_CLOCK_0_IRQ_0)` be sufficient? Especially for a hack.

**Review by galak — APPROVED** [review]

**Comment by aescolar** [comment]
> I don't entirely understand the need for this complex condition check.

No good reason really. But I expect to have DTS support for this board sometime in the next month, so it should be just provisional.

File: boards/posix/nrf52_bsim/hw_models_version [hunk 1]
```diff
@@ -1 +1 @@
-v1.3
\ No newline at end of file
+v1.4
```
Comment (ioannisg) [inline boards/posix/nrf52_bsim/hw_models_version]: Something wrong with the new line character?

**Review by ioannisg — APPROVED** [review]

**Comment by aescolar** [comment]
recheck

**Comment by aescolar** [comment]
recheck

**Pull request merged**